lib/msal-angular/docs/v2-docs/msal-interceptor.md

```diff
@@ -0,0 +1,157 @@
+            ["https://graph.microsoft.com/v1.0/me", ["user.read", "profile"]],
+            ["https://myapplication.com/user/*", ["customscope.read"]]
+        ]),
+    }
+    ```
+
+1. An array of `ProtectedResourceScopes`, which will attach scopes only for specific HTTP methods. 
+
+    ```javascript
+    {
+        interactionType: InteractionType.Redirect,
+        protectedResourceMap: new Map<string, Array<string|ProtectedResourceScopes> | null>([
+            ["https://graph.microsoft.com/v1.0/me", ["user.read"]],
+            ["http://myapplication.com", [
+                {
+                    httpMethod: "POST",
+                    scopes: ["write.scope"]
+                }
+            ]]
+        ])
+    }
+    ```
+
+    Note that scopes for a resource can contain a combination of strings and `ProtectedResourceScopes`. In the below example, a `GET` request will have the scopes `"all.scope"` and `"read.scope"`, whereas as `PUT` request would just have `"all.scope"`.
+
+    ```javascript
+    {
+        interactionType: InteractionType.Redirect,
+        protectedResourceMap: new Map<string, Array<string|ProtectedResourceScopes> | null>([
+            ["http://myapplication.com", [
+                "all.scope",
+                {
+                    httpMethod: "GET",
+                    scopes: ["read.scope"]
+                },
+                {
+                    httpMethod: "POST",
+                    scopes: ["info.scope"]
+                }
+            ]]
+        ])
+    }
+    ```
+
+1. A scope value of `null`, indicating that a resource is to be unprotected and will not get tokens. Resources not included in the `protectedResourceMap` are not protected by default. Specifying a particular resource to be unprotected can be useful when some routes on a resource are to be protected, and some are not. Note that the order in `protectedResourceMap` matters, so null resource should be put before any similar base urls or wildcards.
+
+    ```javascript
+    {
+        interactionType: InteractionType.Redirect,
+        protectedResourceMap: new Map<string, Array<string> | null>([
+            ["https://graph.microsoft.com/v1.0/me", ["user.read", "profile"]],
+            ["https://myapplication.com/unprotected", null]
+            ["https://myapplication.com", ["custom.scope"]]
+        ]),
+    }
+    ```
+
+Other things to note regarding the `protectedResourceMap`:
+
+* **Wildcards**: `protectedResourceMap` supports using `*` for wildcards. When using wildcards, if multiple matching entries are found in the `protectedResourceMap`, the first match found will be used (based on the order of the `protectedResourceMap`). 
+* **Relative paths**: If there are relative resource paths in your application, you may need to provide the relative path in the `protectedResourceMap`. This also applies to issues that may arise with ngx-translate. Relative paths should not have a leading slash.
```

Comment:
"Relative paths should not have a leading slash." Is this always true? I feel like I've seen scenarios where a leading slash was needed.

Reply:
Is this a decision we made or a limitation of a dependency? If it's the former I think we should mandate the opposite for consistency (we require a leading slash for redirectUris/postLogoutRedirectUris if you provide a relative url)

Reply:
Looking back at customers who have experienced using relative paths, it looks like a leading slash works for some but not for others. We don't know what the relative paths in their apps are going to be, but putting a relative path without a leading slash in the `protectedResourceMap` will work for endpoints with or without a leading slash. 
(i.e. `relative` in the PRM will work for `/relative` and `relative` endpoints, but `/relative` in the PRM will not work for `relative` endpoints).

Reply:
I've updated the guidance on this.